### DISCUSSION TITLE
Less agressive notelock

### DISCUSSION THREAD

**Discussion by swekka** [ppy/osu#14881] [Ideas] [answered]
## How it works
I have an idea for a less agressive notelock. It works as follows: if the timing windows of two hit objects overlap and
the first object of those two has not been judged yet (hit or missed) and the player hits the second hit object, the first hit object counts as a miss and the second gets judged normally. The texture that is shown when the first note gets missed should be different from the normal miss texture, because if the player uses a skin that hides 300's, they will only see a miss even though they also cleared a note.

## Comparison
The numbered sentences below correspond to the numbers in the table and the rest of this post, the numbers between brackets are things I'm not really sure about.
1\. It prevents player from clearing hit objects out of order
2\. It eliminates the need for notelock toggle
3\. It eliminates the following problem: "Notelock only becomes a problem if the player is unable to return to the previous object and interact with it properly, thus removing the lock. In this case, a locked note may cause a cascading failure, forcing the player to miss more and more subsequent objects, until they run out of health and fail the beatmap."
(4.) On higher and faster difficulties, it prevents players from losing the rhythm and going off-sync
(5.) It doesn't let players ignore a part of the beatmap

| # | Current notelock | This idea | No notelock |
| - | - | - | - |
| 1. | ✔️ | ✔️ | ❌ |
| 2. | ❌ | ✔️ | N/A |
| 3. | ❌ | ✔️ | ✔️ |
| (4.) | ✔️ | ❌ | ❌ |
| (5.) | ✔️ | ❌ | ❌ |

I included 4 and 5 because the wiki mentioned them but I personally don't fully agree with them.
The main benefit of this idea is 3 and an additional benefit is that this idea eliminates the need for a [notelock toggle](https://twitter.com/ppy/status/1141991534411145216).

**Comment by peppy** [comment, marked as answer]
Have you played lazer? This sounds like what we are already using.

**Reply by swekka** [reply]
i watched an osu!stable replay where i had notelock in osu!lazer and since a replay stores coordinates and keyboard input i thought it would use the new notelock but it didn't.
now i actually played the map and you are right
